### DISCUSSION TITLE
DEV weatherservices.py

### DISCUSSION THREAD

**Discussion by hoetzgit** [roe-dl/weewx-DWD#43] [General]
@roe-dl 

Hallo, 
ich habe an `weatherservices.py` gearbeitet. Wenn Du Interesse und Zeit hast, würde ich mich freuen, wenn Du Dir diesen [Branch](https://github.com/hoetzgit/weewx-DWD/blob/dev/bin/user/weatherservices.py) ansiehst und dazu idealerweise Anmerkungen hast.

Danke und Grüße
Henry

**Comment by hoetzgit** [comment]
@roe-dl 

Danke für Deine Hinweise, habe diese - hoffentlich richtig - umgesetzt.

Grüße
Henry

**Comment by hoetzgit** [comment]
@roe-dl 

Hallo Johanna, hast Du Interesse an einem PR? Ich weiß nicht, ob meine Änderungen für Dich sinnvoll erscheinen und in Deine Konzepte passen. Ich würde es gerne wieder "in die richtigen Hände" geben wollen. Kannst ja mal schauen, ansonsten bleibt es einfach hier "geparkt". Bei mir läuft `weatherservices.py` so in dieser Version.
Henry

**Reply by roe-dl** [reply]
Ich sehe es mir an und gebe dann Rückmeldung.

**Reply by roe-dl** [reply]
Ich glaube, ich sehe nicht mehr durch, was sich alles geändert hat. Mach mal den PR. Daran kann man die Unterschiede leichter sehen und diskutieren. Und wenn Du an Hand der Diskussion etwas änderst, wird es automatisch in den PR übernommen.

**Reply by roe-dl** [reply]
Danke für den PR. Es dauert aber noch etwas, bis ich mir das ansehen kann. Ich habe gerade etwas zu viel auf dem Tisch.

**Reply by roe-dl** [reply]
Ich kommentiere den Punkt mal hier, weil durch die Commits die anderen Kommentarstränge nicht mehr sichtbar sind.

Es geht um `weewx.units.obsgoup_dict.setdefault(obstype, obsgroup)`. Für mich gibt es überhaupt keinen Sinn, eine Meßgröße zu definieren, die man selbst gar nicht braucht. Wir verwenden nur die Namen mit dem Präfix. Warum also überhaupt diese Definition? Ich verstehe es nicht.

`location` vs. `station`: Da würde ich mich gern mal durchsetzen und `location` behalten.

`None` vs. leerer String im `ValueTuple`: Wenn es so ist, daß `None` korrekt ist und von WeeWX verarbeitet wird, dann ist `None` korrekt.

Über das "10_minutes" am Ende der Einheit denke ich nochmal nach. Bis jetzt ist es nur das Gefühl, daß es komisch aussieht. Aber das ist keine Begründung.

**Reply by hoetzgit** [reply]
> Ich kommentiere den Punkt mal hier, weil durch die Commits die anderen Kommentarstränge nicht mehr sichtbar sind.

Bei mir sind alle Kommentare sichtbar. Ich hoffe ich mache nichts falsch. Die Form der Kommunikation innerhalb eines PR und auch die Anpassung eines PR ist für mich neu.

> Es geht um `weewx.units.obsgoup_dict.setdefault(obstype, obsgroup)`. Für mich gibt es überhaupt keinen Sinn, eine Meßgröße zu definieren, die man selbst gar nicht braucht. Wir verwenden nur die Namen mit dem Präfix. Warum also überhaupt diese Definition? Ich verstehe es nicht.

Dafür verstehe ich Dich inzwischen, glaube ich jedenfalls ;-) 

> `location` vs. `station`: Da würde ich mich gern mal durchsetzen und `location` behalten.

Ja natürlich. Hatte nur versucht Dich mit Argumenten umzustimmen, jetzt bin ich ruhig.

> `None` vs. leerer String im `ValueTuple`: Wenn es so ist, daß `None` korrekt ist und von WeeWX verarbeitet wird, dann ist `None` korrekt.

Ich denke schon. Ich habe mir die Kommentare der `unit.py`/`convert` Funktion angesehen.

> Über das "10_minutes" am Ende der Einheit denke ich nochmal nach. Bis jetzt ist es nur das Gefühl, daß es komisch aussieht. Aber das ist keine Begründung.

ja, sieht komisch aus. Wie macht man das generell wenn ein externer Wert in einer Einheit vorliegt, die WeeWX nicht kennt? Aktuell hast Du z.B. in DWDCDCthread `GS_10` so definiert: `('radiation','J/cm^2','group_radiation')`. Damit kann doch WeeWX nicht's anfangen oder? `J/cm^2` wäre dann eigentlich das Label für eine Einheit? Also habe ich versucht, eine Einheit zu definieren.  Dann noch die Konvertierungsfunktion in eine vorhandene WeeWX Einheit für die `radiation`. Jetzt kommt aber noch der Zeitfaktor, wie bringt man zum Ausdruck, dass dieser Wert z.B. die Summe aus 10 Minuten darstellt?

**Reply by hoetzgit** [reply]
Hallo Johanna, aktuell bin ich selbst etwas verwirrt in meinem Code. Es wäre besser, wenn Du den PR ignorierst und schließt. Ich warte lieber bis Du das vernünftig weiterentwickelst. War eine schöne Übung und ich hab auch was gelernt. Für mich aber nun "Schuster, bleib bei deinen Leisten".

**Reply by roe-dl** [reply]
Das Gefühl hatte ich auch gehabt, als ich den ersten PR bei Tom Keffer eingereicht habe. Als das durch war, hatte ich erst einmal genug vom Programmieren. Du bist also nicht allein.

Als völlig unproblematisch würde ich den zusätzlichen Provider sehen, weil Du das allein in der Hand hast. Den können wir auf jeden Fall einfügen. 

Ansonsten sind wir doch fast durch. Du schaffst das.
